### DISCUSSION TITLE
reduce the binary size

### DISCUSSION THREAD

**Discussion by drakkan** [lestrrat-go/jwx#345] [Ideas]
Hi,

first of all thank you for this great library!

After upgrading jwx from v1.1.2 to v1.1.3 the binary size of my app has increased from 33052040 to 34359080 bytes. 

There are now new dependencies to support ES256K signing. Could you please consider moving the supported features into separate sub-packages so that we can only import what we really use? (This will likely be a backward incompatible change). Thank you!

**Comment by lestrrat** [comment]
This is just a PoC and by no means the definitive version, but would you like to try [this branch](https://github.com/lestrrat-go/jwx/tree/topic/toggle-es256k)?

**Reply by drakkan** [reply]
I confirm that it works as expected, the size drops to 33056168 bytes, thank you!

**Reply by lestrrat** [reply]
FYI v1.1.4 has been released https://github.com/lestrrat-go/jwx/releases/tag/v1.1.4

**Reply by drakkan** [reply]
Thank you for the quick fix, I was already using the [latest version](https://github.com/drakkan/sftpgo/blob/main/go.mod#L31). I'll update to 1.1.4 :)
